crates/samples/echomain/src/app.rs

```diff
@@ -3,183 +3,116 @@
 // Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
 // ------------------------------------------------------------
 
-#![allow(non_snake_case)]
-
 use std::cell::Cell;
-use std::convert::TryInto;
-use std::io::Error;
-use std::thread::JoinHandle;
 
 use log::info;
-use mssf_com::FabricCommon::FabricRuntime::{
-    IFabricRuntime, IFabricStatelessServiceFactory, IFabricStatelessServiceFactory_Impl,
-    IFabricStatelessServiceInstance, IFabricStatelessServiceInstance_Impl,
-    IFabricStatelessServicePartition,
-};
-use mssf_com::FabricCommon::{
-    IFabricAsyncOperationCallback, IFabricAsyncOperationContext, IFabricStringResult,
+use mssf_core::runtime::stateless::{
+    PartitionKind, StatelessServiceFactory, StatelessServiceInstance, StatelessServicePartition,
 };
-use mssf_core::{AsyncContext, StringResult};
+use mssf_core::HSTRING;
+use tokio::runtime::Handle;
 use tokio::sync::oneshot::{self, Sender};
-use windows::core::implement;
-use windows::core::w;
-use windows_core::HSTRING;
-
-mod echo;
-
-pub fn run(runtime: &IFabricRuntime, port: u32, hostname: HSTRING) {
-    info!("port: {}, host: {:?}", port, hostname);
+use tokio::sync::Mutex;
+use tokio::task::JoinHandle;
 
-    let factory: IFabricStatelessServiceFactory = ServiceFactory::new(port, hostname).into();
-    let service_type_name = w!("EchoAppService");
-    unsafe { runtime.RegisterStatelessServiceFactory(service_type_name, &factory) }
-        .expect("register failed");
-}
+use crate::echo;
 
-#[derive(Debug)]
-#[implement(IFabricStatelessServiceFactory)]
-pub struct ServiceFactory {
-    port_: u32,
-    hostname_: HSTRING,
+pub struct Factory {
+    port: u32,
+    hostname: HSTRING,
+    rt: Handle,
 }
 
-impl ServiceFactory {
-    pub fn new(port: u32, hostname: HSTRING) -> ServiceFactory {
-        ServiceFactory {
-            port_: port,
-            hostname_: hostname,
-        }
+impl Factory {
+    pub fn new(port: u32, hostname: HSTRING, rt: Handle) -> Self {
+        Self { port, hostname, rt }
     }
 }
 
-impl IFabricStatelessServiceFactory_Impl for ServiceFactory {
-    #[allow(clippy::not_unsafe_ptr_arg_deref)]
-    fn CreateInstance(
+impl StatelessServiceFactory for Factory {
+    fn create_instance(
         &self,
-        servicetypename: &::windows::core::PCWSTR,
-        servicename: *const u16,
-        initializationdatalength: u32,
-        initializationdata: *const u8,
-        partitionid: &::windows::core::GUID,
+        servicetypename: &HSTRING,
+        servicename: &HSTRING,
+        initializationdata: &[u8],
+        partitionid: &mssf_core::GUID,
         instanceid: i64,
-    ) -> ::windows::core::Result<IFabricStatelessServiceInstance> {
-        let mut init_data: String = "".to_string();
```

Comment:
what is the use-case of initialization-data ? is this code moved under safe wrapper or are we completely removing it.

Reply:
Old code is using raw ptr and buffer length:
```rs
 initializationdatalength: u32,
        initializationdata: *const u8,
```
new code is using a slice that has the buffer ref and length inside.
```rs
initializationdata: &[u8],
```
The functionality is the same, but wrapped.